api/addjob_test.go

```diff
@@ -61,8 +73,13 @@ func TestFailureToAddJob(t *testing.T) {
 
 				calls := auditorMock.RecordCalls()
 				So(len(calls), ShouldEqual, 2)
-				testapi.VerifyAuditorCalls(calls[0], addJobAction, audit.Attempted, nil)
+				testapi.VerifyAuditorCalls(calls[0], addJobAction, audit.Attempted, attemptedAuditParams)
 				testapi.VerifyAuditorCalls(calls[1], addJobAction, audit.Unsuccessful, nil)
+
+				Convey("Then the request body has been drained", func() {
+					_, err = r.Body.Read(make([]byte, 1))
```

Comment:
Probably want to check first variable instead of automatically throwing away - `if there is no error you may want to know whether the body is nil or not as this will give you more to work with if it breaks`

Reply:
I have updated the tests to also check the first returned value - the number of bytes read.